packages/prime/src/prime_cli/commands/env.py

```diff
@@ -1927,6 +1926,106 @@ def delete(
         raise typer.Exit(1)
 
 
+def _check_environment_installed(env_name: str, required_version: str | None = None) -> bool:
+    """Check if an environment package is installed and loadable."""
+    try:
+        if required_version:
+            pkg_name = env_name.replace("-", "_")
+            check_code = (
+                f"from verifiers import load_environment; "
+                f"from importlib.metadata import version as get_version; "
+                f"from packaging.version import Version; "
+                f"load_environment('{env_name}'); "
+                f"installed = Version(get_version('{pkg_name}')); "
+                f"required = Version('{required_version}'); "
+                f"exit(0 if installed == required else 1)"
+            )
+        else:
+            check_code = f"from verifiers import load_environment; load_environment('{env_name}')"
+        result = subprocess.run(
+            ["uv", "run", "python", "-c", check_code],
+            capture_output=True,
+            text=True,
+        )
+        return result.returncode == 0
+    except Exception:
+        return False
+
+
+def _auto_install_environment(env_slug: str) -> bool:
```

Comment:
it seems like we're rewriting much of the logic that's already in the `install` function, but missing some additional checks and fallback logic